proposals/0156-high-level-interface-foundation.md

```diff
@@ -34,387 +37,248 @@ The high level interface sits on top of the management layer. It should provide
+Similar to [`UIApplicationMain()`](https://developer.apple.com/documentation/uikit/1622933-uiapplicationmain?language=objc) the function `SDLApplicationMain()` should become **the** entry point for SDL apps. The app developer should call the function in an early phase of the app's launching procedure (e.g. in `UIApplicationDelegate.application:didFinishLaunchingWithOptions:`). It creates the shared application object as well as the object of the class implementing the application delegate and makes the SDL application ready using the app configuration and the named root view controller.
 
-@end
+```objc
+// assume all nullable
+extern void SDLApplicationMain(NSString *applicationClassName, NSString *delegateClassName, NSString *rootClassName, SDLConfiguration *configuration);
+```
 
-@interface SDLVoiceInteraction
+As a start to demonstrate the ease of SDL the function should be able to build the application as HelloSDL by calling `SDLApplicationMain(nil, nil, nil, nil)`. The transport layer should be iAP so it's expected that `ExternalAccessory.framework` linked and the protocol strings are added. The idea of helloSDL in the framework is to provide a very quick start into SDL. It could be added to the iOS documentation as a checkpoint to get an SDL app running.
 
-@property (nonatomic, copy) NSArray<SDLTTSChunk *> *helpPrompt;
-@property (nonatomic, copy) NSArray<SDLTTSChunk *> *timeoutPrompt;
+*Note: There are more variants/alternatives for `SDLApplicationMain()` in the sub-section below.*
 
-@property (nonatomic, copy) NSArray<SDLVrHelpItem *> *helpItems;
+#### applicationClassName
 
-@end
-```
+By default `SDLApplicationMain()` creates the shared application object of `SDLApplication`. The app developer can also provide a named subclass of `SDLApplication` if this is preferred. Most apps don't need this but it would follow `UIApplicationMain()` and allows custom events or access areas beyond the public API. Another example would be to make the subclass implement the delegate.
 
-### Views
+#### delegateClassName
 
-The high level interface introduces three different kind of views: text view, image view and button view. Each view element added to a view controller will be responsible for a parameter of the `Show` RPC.
+The name of the class that implemented the SDL application delegate protocol. If the application subclass implements the delegate this parameter should match `applicationClassName`. Set nil if no delegate is needed or notifications from `NSNotificationCenter` are preferred.
 
-#### SDLView
+#### rootClassName
 
-```objc
-@interface SDLView
+A named subclass of `SDLViewController` to be used as the root view controller for the application. An object of it will be created as soon as the application is launched. Set to `"SDLViewController"` or nil to let the application create a default view controller.
 
-@property (nonatomic) NSArray<SDLView *> *subviews;
+#### configuration
```

Comment:
This should be capitalized "Configuration"

Reply:
Actually that's code in the title (parameter name). I ``'d it.